core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2Metadata.java

```diff
@@ -130,7 +147,64 @@ public String toString() {
      */
     @Override
     public String get(String path) {
-        return "IMDS";
+
+        String data = null;
+        AbortableInputStream abortableInputStream = null;
+        try {
+            String token = getToken();
+            URI uri = URI.create(endpoint + path);
+            HttpExecuteRequest httpExecuteRequest = requestMarshaller.createDataRequest(uri , SdkHttpMethod.GET, token,
+                                                                                tokenTtl);
+            HttpExecuteResponse response =  httpClient.prepareRequest(httpExecuteRequest).call();
+            int statusCode = response.httpResponse().statusCode();
+
+            if (statusCode == HttpURLConnection.HTTP_OK) {
+                abortableInputStream = response.responseBody().get();
+                data = IoUtils.toUtf8String(abortableInputStream);
+            } else if (statusCode == HttpURLConnection.HTTP_NOT_FOUND) {
+                throw SdkServiceException.builder().message("The requested metadata is not found ").build();
+            } else {
+                throw SdkClientException.builder()
+                                        .message("Unexpected Exception occurred with Status Code " + statusCode).build();
```

Comment:
Maybe "Instance metadata service returned unexpected status code X"?
The wording as it is here makes me think that there's an exception we should have logged.